web/app/js/components/BaseTable.test.js

```diff
@@ -82,4 +82,37 @@ describe("Tests for <BaseTable>", () => {
     expect(emptyCard).toBeDefined();
     expect(emptyCard).toHaveLength(1);
   });
+
+  it("if enableFilter is true, user can filter rows by search term", () => {
+    let extraProps = _merge({}, defaultProps, {
+      tableRows: [{
+        deployment: "authors",
+        namespace: "default",
+        key: "default-deployment-authors",
+        pods: {totalPods: "1", meshedPods: "1"}
+      },
+      {
+        deployment: "books",
+        namespace: "default",
+        key: "default-deployment-books",
+        pods: {totalPods: "2", meshedPods: "1"}
+      }],
+      tableColumns: tableColumns,
+      enableFilter: true
+    });
+
+    const component = mount(<BaseTable {...extraProps} />);
+    const enableFilter = component.prop("enableFilter");
+    const filterIcon = component.find("FilterListIcon");
+    expect(enableFilter).toEqual(true);
+    expect(filterIcon).toHaveLength(1);
+
+    filterIcon.simulate("click");
```

Comment:
For my own understanding: It looks like in this test, we are simulating a click on the `filterIcon` to test the filter functionality, but in `MetricsTable.test.js` we are simply changing `filterBy` in the state to test the filter functionality. Is there a reason to test these in different ways?

Reply:
Yes, I didn't want to remove the existing tests, but I hadn't access to button from `MetricsTable` due to `Hidden` component issue, so I decided to keep the existing tests, but in these cases only the state and props of some children components is tested, instead of testing their events and methods.

Reply:
I see, thanks for the explanation! 👍